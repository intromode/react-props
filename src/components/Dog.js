import React from 'react';
import PropTypes from 'prop-types';

export default function Dog({ age, name, weight }) {
  return (
    <dl>
      <dt>Name</dt>
      <dd>{name}</dd>

      <dt>Age</dt>
      <dd>{age}</dd>

      <dt>Weight</dt>
      <dd>{weight}</dd>
    </dl>
  );
}

Dog.propTypes = {
  age: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  weight: PropTypes.string.isRequired,
};
