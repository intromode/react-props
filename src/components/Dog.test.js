import React from 'react';
import { shallow } from 'enzyme';
import Dog from './Dog';

describe('tests dog component', () => {
  it('matches dog screenshot', () => {
    const wrapper = shallow(<Dog age={5} name="snuggles" weight="60 lbs" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
