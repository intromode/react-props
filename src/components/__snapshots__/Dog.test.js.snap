// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests dog component matches dog screenshot 1`] = `
<dl>
  <dt>
    Name
  </dt>
  <dd>
    snuggles
  </dd>
  <dt>
    Age
  </dt>
  <dd>
    5
  </dd>
  <dt>
    Weight
  </dt>
  <dd>
    60 lbs
  </dd>
</dl>
`;
